docs/pages/blog/callback-support-in-style-overrides.md

```diff
@@ -0,0 +1,189 @@
+description: We're excited to introduce the callback support for global theme overrides in this minor version update!
+date: 2022-01-20T00:00:00.000Z
+authors: ['siriwatknp']
+tags: ['MUI Core', 'News']
+card: false
+---
+
+MUI Core [v5.3.0](https://github.com/mui-org/material-ui/releases/tag/v5.3.0) introduces the ability to write a callback in style overrides (the global theming), giving you full control of component customization at the theme level.
+
+Why using a callback is better than the existing plain object? Let me explain from the beginning.
+
+## The problems
+
+In v4, the style engine library was JSS which had some limitations.
+Style overrides were not able to support dynamic props via a callback so we relied on using classes. Take a look at the [`Chip` classes](https://github.com/mui-org/material-ui/blob/97d32b0ff3fae4537c20c79e619f132f4a5c5cbb/packages/mui-material/src/Chip/chipClasses.ts) for example, there are more than 20 classes that are incomplete if we count the permutation of elements (`root | avatar | icon | label | deleteIcon`), size (`small | medium | large`), and color (`primary | secondary | ...`).
+This leads to a bad theming experience because developers need to know the specific key to customize.
+
+We believe it would be better for developers if they could create custom styles by reading the component props, without ever needing to know what key they should use.
+
+Fortunately, it is now possible in v5 because of the new style engine powered by emotion. Theming is simpler and more flexible. You only need to know the component's slot name and then provide an **object** (static overrides) or a **callback** (dynamic overrides).
+
+## Using callback in `styleOverrides`
+
+The callback gives you the `props` that the slot received. Most of the time, you would use:
+
+- `props.ownerState`: the combination of runtime props and internal states.
+- `props.theme`: the object you provide to `ThemeProvider` or the default one.
+
+```jsx
+import { ThemeProvider, createTheme } from '@mui/material/styles';
+
+<ThemeProvider
+  theme={createTheme({
+    components: {
+      MuiChip: {
+        styleOverrides: {
+          // you can now use the theme without creating the initial theme!
+          root: ({ ownerState, theme }) => ({
+            padding: {
+              small: '8px 4px',
+              medium: '12px 6px',
+              large: '16px 8px',
+            }[ownerState.size],
+            ...(ownerState.variant === 'outlined' && {
+              borderWidth: '2px',
+              ...(ownerState.variant === 'primary' && {
+                borderColor: theme.palette.primary.light,
+              }),
+            }),
+          }),
+          label: {
+            padding: 0,
+          },
+        },
+      },
+    },
+  })}
+>
+  ...your app
+</ThemeProvider>;
+```
+
+> 💡 The side benefit of using a callback is that you can use the runtime theme without creating the outer scoped variable.
```

Comment:
> the outer scoped variable

Will developer know what this means? (I might just be missing some context.)

Reply:
My point is some developers might do this:

```js
const theme = createTheme();

<ThemeProvider theme={createTheme({
  components: {
    MuiButton: {
      styleOverrides: {
        root: {
          // just to give an example where I want to access theme in styleOverrides.
          // I have to create theme variable at the outer scope.
          backgroundColor: theme.palette.common.black,
        }
      }
    }
  }
})}>
```

Reply:
> Will developer know what this means?

Good point, I guess not everyone would understand. I add it without an actual code sample because I think it is just a side benefit which is not the main purpose.

Reply:
> I think it is just a side benefit which is not the main purpose.

We have multiple threads where developers complain about this. But from what I remember, it was a step before. It was for building the core set of design tokens, not so much for customization of components.